protos/shell/shell.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package mavsdk.rpc.shell;
+
+option java_package = "io.mavsdk.shell";
+option java_outer_classname = "ShellProto";
+
+/**
+ * Shell Service allow to communicate with vehicle's system shell.
+ */
+service ShellService {
+    // Communicate with a vehicle's Shell.
+    rpc SetShellMessage(SetShellMessageRequest) returns(SetShellMessageResponse) {}
+}
+
+message SetShellMessageRequest {
+    ShellMessage shell_message = 1;
+}
+message SetShellMessageResponse {
+    ShellResult shell_result = 1;
+}
+
+message ShellMessage {
+    bool need_response = 1; // Set if the sender wants the receiver to send a response.
+    uint32 timeout_ms = 2; // Timeout (ms).
+    string data = 3; // Serial data (70 symbols max).
+}
+
+message ShellResult {
+    // Possible results returned for shell requests
+    enum Result {
+        UNKNOWN = 0; // Unknown error
+        SUCCESS = 1; // Request succeeded
+        NO_SYSTEM = 2; // No system is connected
+        CONNECTION_ERROR = 3; // Connection error
+        DATA_TOO_LONG = 4; // Request Data too long
+        NO_RESPONSE = 5; // Response does not received
+    }
+
+    Result result = 1; // Result enum value
+    string result_str = 2; // Human-readable English string describing the result
+    string response_data = 3; // Response message (if available)
```

Comment:
This should go in `SetShellMessageResponse` (lines 19-21). What you put in `ShellResult` is used to generate an exception (e.g. in Python, if the result is not `SUCCESS`, it will raise an exception).

Reply:
```python
[+] Generating plugins from 
 -> [+] Generated protobuf and gRPC bindings for shell
Traceback (most recent call last):
  File ".venv/bin/protoc-gen-dcsdk", line 11, in <module>
    load_entry_point('protoc-gen-dcsdk==0.1', 'console_scripts', 'protoc-gen-dcsdk')()
  File ".venv/lib/python3.6/site-packages/protoc_gen_dcsdk/__main__.py", line 17, in main
    AutoGen.generate_reactive(request).SerializeToString())
  File ".venv/lib/python3.6/site-packages/protoc_gen_dcsdk/autogen.py", line 77, in generate_reactive
    f.content = str(out_file)
  File ".venv/lib/python3.6/site-packages/protoc_gen_dcsdk/autogen_file.py", line 34, in __repr__
    has_result=self._has_result)
  File ".venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File ".venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File ".venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File ".venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "mavsdk-python/other/tools/../..//other/templates/file.j2", line 54, in top-level template code
    {{ '\n' + indent(method, 1) }}
  File ".venv/lib/python3.6/site-packages/protoc_gen_dcsdk/utils.py", line 83, in jinja_indent
    _in_str = str(_in_str)
  File ".venv/lib/python3.6/site-packages/protoc_gen_dcsdk/methods.py", line 220, in __repr__
    has_result=self._has_result)
  File ".venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File ".venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File ".venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File ".venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "mavsdk-python/other/tools/../..//other/templates/request.j2", line 40, in top-level template code
    request.{{ param_name.lower_snake_case }} = param.name.rpc()
  File ".venv/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'param_name' is undefined
--custom_out: protoc-gen-custom: Plugin failed with status code 1.
mv: cannot stat 'mavsdk-python/other/tools/../..//mavsdk/generated/Shell.py': No such file or directory
```

Reply:
Oh, there is a typo in the template. Instead of `param_name`, it should be `param.name` [here](https://github.com/mavlink/MAVSDK-Python/blob/master/other/templates/request.j2#L40). Would you mind trying that?

Reply:
Not at all.
Produced code:
```python
request = shell_pb2.SetShellMessageRequest()            
request.shell_message = param.name.rpc()         # param not founded
response = await self._stub.SetShellMessage(request)
```

Reply:
Ok, I'll have a look, there is a bug somewhere in our generator.